scripts/nb-tester/qiskit_docs_notebook_tester/config.py

```diff
@@ -0,0 +1,448 @@
+    notebooks_no_mock: list[str]
+
+    @property
+    def all_job_submitting_notebooks(self) -> list[str]:
+        return [*self.notebooks_that_submit_jobs, *self.notebooks_no_mock]
+
+    @property
+    def all_notebooks_to_test(self) -> list[str]:
+        return [
+            *self.notebooks_normal_test,
+            *self.notebooks_that_submit_jobs,
+            *self.notebooks_no_mock,
+        ]
+
+    @property
+    def all_notebooks(self) -> list[str]:
+        return [
+            *self.all_notebooks_to_test,
+            *self.notebooks_exclude,
+        ]
+
+    @classmethod
+    def from_args(cls, args: argparse.Namespace) -> Config:
+        """
+        Create config from args, including loading the globs from the TOML file
+        """
+        path = Path(args.config_path)
+        try:
+            return cls(args=args, **tomli.loads(path.read_text()))
+        except TypeError as err:
+            raise ValueError(
+                f"Couldn't read config from {path}; check it exists and the"
+                " entries are correct."
+            ) from err
+
+    def notebooks_to_execute(self) -> Iterator[Path]:
+        """
+        Yield notebooks to be executed, printing messages for any skipped files.
+        """
+        paths = map(Path, self.args.filenames or self.all_notebooks_to_test)
+        for path in paths:
+            if path.suffix != ".ipynb":
+                print(f"ℹ️ Skipping {path}; file is not `.ipynb` format.")
+                continue
+
+            if matches(path, self.notebooks_exclude):
+                print(
+                    f"ℹ️ Skipping {path}; to run it, edit `notebooks-exclude` in {self.args.config_path}."
+                )
+                continue
+
+            if not self.args.submit_jobs and matches(path, self.notebooks_no_mock):
+                print(
+                    f"ℹ️ Skipping {path} as it doesn't work with mock hardware; use the --submit-jobs flag to run it."
+                )
+                continue
+
+            if self.args.only_submit_jobs and not matches(
+                path, self.all_job_submitting_notebooks
+            ):
+                print(
+                    f"ℹ️ Skipping {path} as it doesn't submit jobs and the --only-submit-jobs flag is set."
+                )
+                continue
+
+            yield path
+
+    def should_patch(self, path: Path) -> bool:
+        if self.args.submit_jobs:
+            return False
+        return matches(path, self.notebooks_that_submit_jobs)
+
+    def should_skip_writing(self, path: Path) -> bool | str:
+        """Returns False or string with reason for skipping"""
+        if not self.args.write:
+            return "--write arg not set"
+        if self.should_patch(path):
+            return "hardware was mocked"
+        return False
+
+
+def matches(path: Path, glob_list: list[str]) -> bool:
+    return any(path.match(glob) for glob in glob_list)
+
+
+
+def get_args() -> argparse.Namespace:
```

Comment:
Maybe we can keep this function in `__init__.py` to reduce the complexity of the `config.py` script. Although, I think it's fine either way if you like it more here :)

Reply:
I think it makes sense here since it's about config